### PR TITLE
Add debugging support

### PR DESCRIPTION
**Pull request by danebou**
Well, I just noticed there's already another feature branch. But here's the full feature regardless.

### CODE REVIEW

**Comment by numaru** [comment]
Hello and thanks for contributing. I was not happy with the first implementation of the debug feature and almost forgot about it.

**Comment by danebou** [comment]
Thanks! It will be really nice being able to debug some of our systems without needing hardware. I'm new to typescript, so feel free to point out anything odd. Hopefully this might be a bit less hacky.

File: README.md [hunk 1]
```diff
@@ -45,6 +46,38 @@ ID                                 | Command
...
+
+Note: Individual test debugging is not supported. Instead the entire test file will be ran, so skip or remove breakpoints accordingly.
+
+Example configuration with Native Debug (`webfreak.debug`):
+```json
+{
+    "name": "Ceedling Test Explorer Debug",
+    "type": "cppdbg",
+    "request": "launch",
+    "program": "Ceedling/build/test/out/${command:ceedlingExplorer.debugTestExecutable}",
```
Comment (numaru) [inline README.md]:
```suggestion
    "program": "${workspaceFolder}/build/test/out/${command:ceedlingExplorer.debugTestExecutable}",
```

File: README.md [hunk 2]
```diff
@@ -45,6 +46,38 @@ ID                                 | Command
 `test-explorer.run-test-at-cursor` | Run the test at the current cursor position
 `test-explorer.cancel`             | Cancel running tests
 
+## Debugging
+
+To set up debugging, create a new Debug Configuration. `${command:ceedlingExplorer.debugTestExecutable}` 
+can be used access the .out test executable filename (without directory) being ran. Depending on your Ceedling configuration
+these can be found in `Ceedling/build/test/out/`. Then, edit the `ceedlingExplorer.debugConfiguration` settings with the name of the Debug Configuration to run during debug.
```
Comment (numaru) [inline README.md]: The ceedling build is not always in a `Ceedling` directory.

```suggestion
these can be found in `projectPath/build/test/out/`. Then, edit the `ceedlingExplorer.debugConfiguration` settings with the name of the Debug Configuration to run during debug.
```

**Comment by danebou** [comment]
Good catch on the package.json. Forgot to check that in.

**Pull request merged**